src/main/jbake/content/blog/2017/01/11/hawkular-metrics-with-dropwizard.adoc

```diff
@@ -0,0 +1,329 @@
+    hawkularReporter.start(1, TimeUnit.SECONDS);
+```
+
+And that's it. From now on, all metrics that is added to the registry will be reported to Hawkular Metrics.
+The default configuration expects to find the Hawkular server on _localhost:8080_. You can explore the builder options to configure it.
+
+TIP: In this example, "com.hawkular.sample" refers to the tenant used in Hawkular. You can set whatever you want instead. link:http://www.hawkular.org/hawkular-metrics/docs/user-guide/#_tenants[Learn more] about Hawkular's multi-tenancy.
+
+If, for instance, we want to connect to an Hawkular pod running in OpenShift, we would configure the reporter similar to that one:
+``` java
+    HawkularReporter hawkularReporter = HawkularReporter.builder(registry, "com.hawkular.sample")
+            .uri("https://metrics.192.168.42.63.xip.io/hawkular/metrics")
+            .bearerToken("ABCDEFGHIJ1234")
+            .build();
+```
+
+To conclude with the initial setup, note that it is often a good practice to include some host-based disambiguation string in the metric names,
+so that if your application is deployed on several hosts and they all communicate to the same Hawkular instance using the same tenant,
+they will not conflict and feed the same metrics.
+A convenient way to do that is to set a prefix in the reporter configuration, for instance using the host name:
+
+``` java
+    String hostname;
+    try {
+        hostname = InetAddress.getLocalHost().getCanonicalHostName();
+    } catch (UnknownHostException e) {
+        hostname = "?";
+    }
+    HawkularReporter hawkularReporter = HawkularReporter.builder(registry, "com.hawkular.sample")
+            .prefixedWith(hostname + ".")
+            .build();
+```
+
+=== Dropwizard -> Hawkular
+
+As stated before, our sample app will do some (very) basic benchmarking on several caches implementations: we'll have a _Guava_ cache and a local _EhCache_.
+
+NOTE: This sample app is on my GitHub: https://github.com/jotak/hawkular-dropwizard-sample
+
+We create a very simple interface that allows us to get data from cache, count number of elements and know if the latest read was from cache or not:
+
+``` java
+interface Backend {
+    Object get(String key);
+    long count();
+    void init(Map<String, Object> presetElements);
+    boolean isLastReadFromCache();
+}
+```
+
+You can link:https://github.com/jotak/hawkular-dropwizard-sample/tree/master/src/main/java/com/hawkular/sample[see on GitHub] the `GuavaBackend` and `EhcacheBackend` implementations, there's nothing fancy here.
+
+EhCache is initialized programmatically:
+
+``` java
+    CacheManager cacheManager = CacheManager.newInstance();
+    Ehcache cache = cacheManager.addCacheIfAbsent("testCache");
+```
+
+Now, the `Benchmark` class. This is where I create metrics and feed them, with the pretext of a small scenario. Here's the constructor:
+
+``` java
+    Benchmark(MetricRegistry registry, Ehcache ehcache) {
+        // (1)
```

Comment:
Those should probably behind the statement line.  See http://asciidoctor.org/docs/user-manual/#copy-and-paste-friendly-callouts